### ISSUE TITLE
Failures when calling class/static method directly on derived class (not on instance)

### ISSUE THREAD

**Issue by adam-sikora** [flexmock/flexmock#72]
I have encountered multiple flexmock fails when mocking class/static methods that are called directly on a derived class. The issues started happening in `0.10.5` and are still present in `0.10.7`. Tests to reproduce the failures:
```
class A:

    @classmethod
    def class_method(cls):
        return 1

    @staticmethod
    def static_method():
        return 1

    @classmethod
    def with_arg(cls, integer):
        return integer

    @staticmethod
    def with_arg_static(integer):
        return integer


class B(A):
    pass


def test_class_method_on_derived_class():
    flexmock(A).should_receive('class_method').and_return(1)
    # passes
    A.class_method()

    flexmock(B).should_receive('class_method').and_return(1)
    # passes
    B().class_method()

    # fails with:
    # TypeError: mock_method() missing 1 required positional argument: 'runtime_self'
    B.class_method()


def test_static_method_on_derived_class():
    flexmock(A).should_receive('static_method').and_return(1)
    # passes
    A.static_method()

    flexmock(B).should_receive('static_method').and_return(1)
    # passes
    B().static_method()

    # fails with:
    # TypeError: mock_method() missing 1 required positional argument: 'runtime_self'
    B.static_method()


def test_class_method_with_args_on_derived_class():
    flexmock(A).should_receive('with_arg').with_args(1).and_return(1)
    # passes
    A.with_arg(1)

    flexmock(B).should_receive('with_arg').with_args(1).and_return(1)
    # passes
    B().with_arg(1)

    # fails with:
    # flexmock.MethodSignatureError: get_one()
    # Did not match expectation get_one(1)
    B.with_arg(1)


def test_static_method_with_args_on_derived_class():
    flexmock(A).should_receive('with_arg_static').with_args(1).and_return(1)
    # passes
    A.with_arg_static(1)

    flexmock(B).should_receive('with_arg_static').with_args(1).and_return(1)
    # passes
    B().with_arg_static(1)

    # fails with:
    # flexmock.MethodSignatureError: with_arg_static()
    # Did not match expectation with_arg_static(1)
    B.with_arg_static(1)
```

**Comment by christophe-riolo** [comment]
~~I can reproduce this issue all the way from 0.7.0.
When testing with Pytest, however, before 0.10.5 I had a failure about the durations. Did you run your tests with Pytest ?
In any case, there is a bug to solve here thanks !~~
My bad, I did not test properly, and I can reproduce, I'll investigate

**Comment by ollipa** [comment]
@christophe-riolo, I wrote tests to our test suite here: https://github.com/flexmock/flexmock/pull/76

You can use those tests to verify your fix.

**Comment by ollipa** [comment]
Closed by #76 and #77
